R/create_task_makefile.R

```diff
@@ -82,23 +83,43 @@ create_task_makefile <- function(
   }), use.names=FALSE)
   job_output <- if(isTRUE(ind_complete) || is.na(ind_complete)) {
     if(is.null(ind_dir)) stop('ind_dir must not be NULL when ind_complete=TRUE')
-    file.path(ind_dir, as_ind_file(job_target, ind_ext))
+    file.path(ind_dir, as_ind_file(target_name, ind_ext))
   } else {
-    sprintf('%s_%s', job_target, ind_ext)
+    sprintf('%s_%s', target_name, ind_ext)
   }
+  
+  # prepare the task list for rendering
+  tasks <- unname(task_plan) # remove list element names where they'd interfere with whisker.render
+  for(task in seq_along(tasks)) {
+    # remove lmore ist element names where they'd interfere with whisker.render
+    tasks[[task]]$steps <- unname(tasks[[task]]$steps)
+    for(step in seq_along(tasks[[task]]$steps)) {
+      # add a logical for has_depends
+      tasks[[task]]$steps[[step]]$has_depends <- length(tasks[[task]]$steps[[step]]$depends) > 0
+    }
+  }
+  
   job <- list(
-    target_name = job_target,
+    target_name = target_name,
```

Comment:
I think there's still room for a non-combiner summary job_target (which stores the file names and hashes of the final step from each task), so I'd rather not repurpose that existing summary target here (or allow the user to choose its name; it's fixed to enforce uniformity across projects). In particular, a job target is useful if you don't need a combiner or haven't written it yet, or if you want to commit that job .ind file so that you can git diff any changes to the output of individual steps.

Could we instead allow the user to specify the target_name[s] of the combiner[s] as other targets, to complement the `job_target` if `ind_complete=TRUE` and to replace the `job_target` if `ind_complete=FALSE`?

Reply:
yes, I agree. Will change